app/controllers/scenarios_controller.rb

```diff
@@ -95,6 +95,16 @@ def update
     end
   end
 
+  def enable_or_disable_all_agents
+    @scenario = current_user.scenarios.find(params[:id])
+    @scenario.agents.update_all disabled: params[:scenario][:disabled]
```

Comment:
The specs are failing on postgres because it handles boolean field differenctly then mysql. I think there are two options, either cast the param value to a ruby boolean here (`!!params[:scenario][:disabled].to_i`or `params[:scenario][:disabled] == 'true' ? true : false`) or change the hidden field to a [hidden checkbox](http://stackoverflow.com/a/1766434) whose value rails automatically casts to a boolean.